apps/studio/src/server/modules/aws/codebuild.service.ts

```diff
@@ -0,0 +1,76 @@
+import type { StartBuildCommandOutput } from "@aws-sdk/client-codebuild"
+import type pino from "pino"
+import {
+  BatchGetBuildsCommand,
+  CodeBuildClient,
+  ListBuildsForProjectCommand,
+  StartBuildCommand,
+  StopBuildCommand,
+} from "@aws-sdk/client-codebuild"
+
+const client = new CodeBuildClient({ region: "ap-southeast-1" })
+
+export const stopRunningBuilds = async (
+  logger: pino.Logger<string>,
+  projectId: string,
+): Promise<void> => {
+  try {
+    // List builds for the given project
+    const listBuildsCommand = new ListBuildsForProjectCommand({
+      projectName: projectId,
+    })
+    const listBuildsResponse = await client.send(listBuildsCommand)
+
+    const buildIds = listBuildsResponse.ids ?? []
+
+    if (buildIds.length === 0) {
+      logger.info({ projectId }, "No running builds found for the project")
+      return
+    }
+
+    // Get details of the builds
+    const batchGetBuildsCommand = new BatchGetBuildsCommand({ ids: buildIds })
+    const batchGetBuildsResponse = await client.send(batchGetBuildsCommand)
+
+    // Stop running builds
+    for (const build of batchGetBuildsResponse.builds ?? []) {
+      if (build.buildStatus === "IN_PROGRESS") {
+        logger.info(
+          { buildId: build.id },
+          "Stopping currently running CodeBuild",
+        )
+        const stopBuildCommand = new StopBuildCommand({ id: build.id })
+        await client.send(stopBuildCommand)
+        logger.info({ buildId: build.id }, "Build stopped successfully")
+      }
+    }
+  } catch (error) {
+    logger.error(
+      { projectId, error },
+      "Unexpected error while stopping running builds",
+    )
+
+    throw error
```

Comment:
actually this works very differently from what this chunk of code expects - ie, a linear seq of events (we `await` various times in this chunk) but worst case doesn't seem that bad (double cancels, double builds) so ok with thi fo rnow

Reply:
By right there will never be a case of double cancels. We always stop running build before starting new one, hence worst case is we await 1x to cancel. a running build